PropertyBinder/Helpers/ExpressionHelpers.cs

```diff
@@ -122,5 +124,96 @@ public static Expression CombineToBlock(params Expression[] expressions)
 
             return Expression.Block(filteredExpressions);
         }
+
+        public static Func<TContext, string> Stamped<TContext>(Expression exp)
```

Comment:
So, does this execute eagerly on every rule creation? This seems harsh performance-wise.
Can we postpone all the work until exception is actually thrown?